### ISSUE TITLE
Add support for Raspberry pi4

### ISSUE THREAD

**Issue by palvarez89** [osmc/osmc#521]
I've recently bought a rbp4 and tried to put OSMC on it. I didn't manage to get it to boot, not even the rainbow screen. Raspbian booted fine.

My guess is that the new version requires some extra/different files in the boot partition and that's why it isn't working.

I can have a look at what's missing to get it all working, but decided to report it here just in case somebody else is working on it.

Ideas, pointers and suggestions welcome!

**Comment by samnazarko** [comment]
See the blog. Support is not ready yet. When it is, it will be announced on the blog.

**Comment by palvarez89** [comment]
Nothing I can do to help then?

**Comment by samnazarko** [comment]
Everything is pretty much done — I just need to arrange a release

**Comment by palvarez89** [comment]
Great news then, thanks for the update

**Comment by NH3CL** [comment]
Any news about Rpi 4 support?

**Comment by samnazarko** [comment]
It will be announced on the blog when released

**Comment by palvarez89** [comment]
Kind of amusing how everything was "pretty much done" 4 months ago, and still hasn't been released.

**Comment by NH3CL** [comment]
Yup, its really strange and frustrating... I thought that they will release OSMC within 1 month after Rpi 4 release....

**Comment by samnazarko** [comment]
It’s open source so I’m happy to accept PRs here to bring things up faster. We will launch instead with a 64-but kernel and armhf userland, which will take a bit more time.

**Comment by NH3CL** [comment]
Do you have ETA for the release? 

5 months has passed since you told us:

> Everything is pretty much done — I just need to arrange a release

**Comment by samnazarko** [comment]
As explained above, I’ve made some changes.

I’ll release it when it’s ready and keep posting updates on the blog.

**Comment by NH3CL** [comment]
You didn't answered any of my questions.... ok, no problem.

**Comment by samnazarko** [comment]
I don’t have a firm ETA at this time.
What are your questions?

**Comment by NH3CL** [comment]
> what do you think, can we get it in this year or only in 2020?

**Comment by samnazarko** [comment]
As explained above, there is no longer a firm ETA at this time. I am not sure how to make that clearer

**Comment by NH3CL** [comment]
No, no no... i've understood everything for the first try. Unfortunately i see that opensource cummunity distributors are going on the wrong way. They are getting more arrogant, they doesn't give proper answers for direct questions ...etc. And that's sad. I would gladly give money for well working softwares (i usually do) and donate for OS community, BUT if the devs are not communicating "clear/right" i wont give any penny.

I had a clear question: "what do you think, can we get it in this year or only in 2020?" And your answer was only bullshit. I would glad with a simple "yes" or "no"....

PS.: no offense.

**Comment by samnazarko** [comment]
I think my answer was pretty clear. I just don't think it's what you wanted to hear.
It's still being worked on.

**Comment by harryberlin** [comment]
don't hassle the sam. he does a great job. LE is in alpha state, too.
if you don't like osmc, so switch to LE.

**Comment by palvarez89** [comment]
> It’s an open source project so I’m happy to accept PRs here to bring things up faster. We will launch instead with a 64-but kernel and armhf userland, which will take a bit more time.

Please, note that I tried to help.

> Nothing I can do to help then? Nothing tracking the work? A WIP branch or something?

I believe I understand how some open source projects work, and I was even excited to know I could help this project migrate to a new board. Just saying, that answering those questions could have made me (and others) engage with the project.

Thanks for all the work done for free, I'm sure we all appreciate it.

**Comment by frankdibs** [comment]
Sam has no interest in adding support for RPi4 because it goes against the now commercial interests of this project and that is pushing people towards the outdated Vero 4K+ that he sells.

**Comment by KODeKarnage** [comment]
> Sam has no interest in adding support for RPi4 because it goes against the now commercial interests of this project and that is pushing people towards the outdated Vero 4K+ that he sells.
Given the Pi has been supported from the Vero 1 through to the Vero 4k+, I don't think it is possible for you to have said anything more stupid. Though I am sure you'll follow up with some valiant effort to try and break that record.
You are the reason open source is dying. We are better off without you.
For the sake of the community, you are hereby banned from using any and all open source software from this point forward until you either die or learn to code and stack up at 100 commits on that shiny new github account you seem to have created for no other purpose than displaying your abject worthlessness.

**Comment by frankdibs** [comment]
> We are better off without you.
I guess the open-source community would be better without OSMC trying to profit on the back of other peoples work also. Thankfully good projects like LibreELEC exist without ulterior motives to support those being neglected here.

**Comment by KODeKarnage** [comment]
> Though I am sure you'll follow up with some valiant effort to try and break that record.

Called it.

**Comment by ActionAdam** [comment]
>pushing people towards the outdated Vero 4K+

Can you show us on the doll where that bad, bad Vero4k touched you?

Seriously, if you are a happy LE user, why are you so triggered? Your actions really make it seem like you are unhappy just because you feel entitled to use OSMC on your pi and are just too impatient to wait. Grow up...

**Comment by frankdibs** [comment]
> Called it.

Expected just as much from a narrow minded person 👍 

> Can you show us on the doll where that bad, bad Vero4k touched you?
> Grow up...

Maybe you should practice what you preach rather than being a hypocrite.

> why are you so triggered

Oh I'm not 😄 ... 5 months on from release when this team had early access to hardware and still no pudding, it's quite clear what those at the top are focused on and what he really cares about.

@NH3CL was bang on point about the arrogance shown by this team.

**Comment by KODeKarnage** [comment]
Stop stop stop, you can't possibly make more of a fool of yourself!

**Comment by frankdibs** [comment]
Sure, soon as you prove me wrong :)

**Comment by KODeKarnage** [comment]
You just proved me wrong.

**Comment by zjoasan** [comment]
Most of developers of OSMC has access to one Vero4k, while that's not the case with the pi4. So time spent with device specific coding is limited to access. Myself don't own a pi4 and my Vero4k negates the need for a new pi for me personally.

**Comment by samnazarko** [comment]
Did you really just create an account to post here?

>Sam has no interest in adding support for RPi4 because it goes against the now commercial interests of this project and that is pushing people towards the outdated Vero 4K+ that he sells.

As soon as you started writing this, I worked out who you were..

Anyone else will know that I have worked on open source for over a decade. We supported the original Apple TV for almost ten years after Apple stopped supporting it. We will support the Raspberry Pi 4 when we are able to do so to a good standard, just as we have been supporting Raspberry Pi for the last seven years. 

Of course, you're not interested in the truth. You've also made us have to close this issue where other users could legitimately enquire about the progress of development.
